packages/messageport/tizen/src/messageport_tizen_plugin.cc

```diff
@@ -0,0 +1,236 @@
+      flutter::EncodableValue value = map[flutter::EncodableValue(key)];
+      if (std::holds_alternative<T>(value)) {
+        out = std::get<T>(value);
+        return true;
+      }
+      LOG_ERROR("Key %s not found", key);
+    }
+    return false;
+  }
+
+  bool GetEncodableValueFromArgs(const flutter::EncodableValue *args,
+                                 const char *key,
+                                 flutter::EncodableValue &out) {
+    if (std::holds_alternative<flutter::EncodableMap>(*args)) {
+      flutter::EncodableMap map = std::get<flutter::EncodableMap>(*args);
+      if (map.find(flutter::EncodableValue(key)) != map.end()) {
+        out = map[flutter::EncodableValue(key)];
+        return true;
+      }
+    }
+    return false;
+  }
+
+  void HandleMethodCall(
+      const flutter::MethodCall<flutter::EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+    LOG_DEBUG("HandleMethodCall: %s", method_call.method_name().c_str());
+    const flutter::EncodableValue *args = method_call.arguments();
+
+    if (method_call.method_name().compare("createLocal") == 0) {
+      CreateLocal(args, std::move(result));
+    } else if (method_call.method_name().compare("createRemote") == 0) {
+      CreateRemote(args, std::move(result));
+    } else if (method_call.method_name().compare("send") == 0) {
+      Send(args, std::move(result));
+    } else {
+      result->Error("Invalid method");
+    }
+  }
+
+  void CreateRemote(
+      const flutter::EncodableValue *args,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+    LOG_DEBUG("CreateRemote");
+    std::string remote_app_id = "";
+    std::string port_name = "";
+    bool is_trusted = false;
+    if (!GetValueFromArgs<std::string>(args, "remoteAppId", remote_app_id) ||
+        !GetValueFromArgs<std::string>(args, "portName", port_name) ||
+        !GetValueFromArgs<bool>(args, "isTrusted", is_trusted)) {
+      result->Error("Invalid parameter");
+      return;
+    }
+
+    bool port_check = false;
+    MessagePortResult native_result = manager.CheckRemotePort(
+        remote_app_id, port_name, is_trusted, &port_check);
+    if (native_result) {
+      result->Success(flutter::EncodableValue(port_check));
+    } else {
+      result->Error("Could not create remote port", native_result.message());
+    }
+  }
+
+  void CreateLocal(
+      const flutter::EncodableValue *args,
+      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
+    LOG_DEBUG("CreateLocal");
+    std::string port_name = "";
+    bool is_trusted = false;
+    if (!GetValueFromArgs<std::string>(args, "portName", port_name) ||
+        !GetValueFromArgs<bool>(args, "isTrusted", is_trusted)) {
+      result->Error("Could not create local port", "Invalid parameter");
+      return;
+    }
+
+    std::stringstream event_channel_name;
+    int id = nextId_++;
```

Comment:
```nextId_``` will be negative someday and you use -1 as initial value at line, it seems to intend invalid state.
I don't think it's gonna be a problem right now, but I don't think you intend to allow negative id

Reply:
`id` from method `Send` will be only used when there is a proper value sent from Dart layer. If next_id_ became -1 (after creating 2^32 ports in one run of application) then nothing wrong should happen.